packages/azure-http-specs/specs/azure/client-generator-core/hierarchy-building/main.tsp

```diff
@@ -0,0 +1,207 @@
+import "@typespec/http";
+import "@typespec/spector";
+import "@azure-tools/typespec-client-generator-core";
+
+using Http;
+using Spector;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for @hierarchyBuilding decorator.")
+@scenarioService("/azure/client-generator-core/hierarchy-building")
+namespace _Specs_.Azure.ClientGenerator.Core.HierarchyBuilding;
+
+@discriminator("kind")
+model Animal {
+    @doc("The kind of animal")
+    kind: string;
+    
+    @doc("Name of the animal")
+    name: string;
+  }
+
+  alias PetContent = {
+    @doc("Whether the pet is trained")
+    trained: boolean;
+  };
+
+  model Pet extends Animal {
+    kind: "pet";
+    ...PetContent;
+  }
+
+  alias DogContent = {
+    @doc("The breed of the dog")
+    breed: string;
+  };
+
+  @global.Azure.ClientGenerator.Core.Legacy.hierarchyBuilding(Pet)
+  model Dog extends Animal {
+    kind: "dog";
+    ...PetContent;
+    ...DogContent;
+  }
+
+interface AnimalOperations {
+  @scenario
+  @scenarioDoc("""
+    Test operation that accepts Animal input and returns Animal output.
+    This validates basic polymorphic behavior in the inheritance hierarchy.
+    Expected request body:
+    ```json
+    {
+      "kind": "animal",
+      "name": "Generic Animal"
+    }
+    ```
+    Expected response body:
+    ```json
+    {
+      "kind": "animal",
+      "name": "Generic Animal"
+    }
+    ```
+    """)
+  @doc("Update an animal as animal")
+  @route("/animal/animal")
+  @put
+  updateAnimalAsAnimal(@body animal: Animal): Animal;
+
+  @scenario
+  @scenarioDoc("""
+    Test operation that accepts Pet input and returns Animal output.
+    This validates that a derived type can be passed where base type is expected.
+    Expected request body:
+    ```json
+    {
+      "kind": "pet",
+      "name": "Buddy",
+      "trained": true
+    }
+    ```
+    Expected response body:
+    ```json
+    {
+      "kind": "pet",
+      "name": "Buddy"
+    }
+    ```
```

Comment:
Now we have multi operations while it is too many and some doesn't follow real case. Let us redefine them into 2 operations:
- `updatePet(@body animal: Animal): Animal;` with route `/animal/pet`. The expected request and expected response shall be same as `Pet`
- `updateDog(@body animal: Animal): Animal;` with route `/animal/dog`. The expected request and expected response shall be same as `Dog`

NOTE: DO update mockapi.ts after change of main.tsp